src/screens/send/send.tsx

```diff
@@ -427,193 +427,291 @@ export class SendScreenComponent extends React.Component<
-                            selectionColor={theme.colors.accent}
-                            value={
-                                this.state.isValidText
-                                    ? formatAddress(this.state.toAddress, account.blockchain)
-                                    : this.state.toAddress
+
+                <View style={styles.content}>
+                    <HeaderStepByStep
+                        steps={[
+                            { title: translate('Send.addAddress') },
+                            {
+                                title: translate('Send.enterAmount'),
+                                selected: this.state.isStep2
+                            },
+                            {
+                                title: translate('Send.confirmTransaction'),
+                                selected: this.state.isStep3
                             }
-                            onChangeText={text => {
-                                this.verifyInputText(text);
-                            }}
-                        />
-                        {this.renderRightAddressIcon()}
-                    </View>
-                    {this.state.errorResponseText && (
-                        <Text style={styles.displayError}>{this.state.errorResponseText}</Text>
-                    )}
-                    {this.state.warningResponseText && (
-                        <Text style={styles.receipientWarning}>
-                            {this.state.warningResponseText}
-                        </Text>
-                    )}
-                    <TouchableOpacity
-                        testID="transfer-between-accounts"
-                        onPress={this.onTransferBetweenAccounts}
-                        style={[styles.buttonRightOptions]}
-                    >
-                        <Text style={styles.textTranferButton}>
-                            {this.state.showOwnAccounts
-                                ? translate('App.labels.close')
-                                : translate('Send.transferOwnAccounts')}
-                        </Text>
-                    </TouchableOpacity>
-
-                    {this.renderAddAddressToBook()}
-
-                    {this.state.userAction && this.renderContinueAction()}
-
-                    {this.state.isValidText && this.renderBasicFields()}
-
-                    {!this.state.isValidText && this.renderListOrBook()}
-                </ScrollView>
+                        ]}
+                        selectPreviousStep={(index: number) => {
+                            if (index === 1 && isStep3) {
+                                this.setState({ isStep2: true, isStep3: false });
+                            } else if (index === 0) {
+                                this.setState({ isStep2: false, isStep3: false });
+                            }
+                        }}
+                    />
```

Comment:
It makes sense here but I would've added and extra component like
SelectToAddressComponent - that is basically the step one for send and will have: Address book, Transfer between account component, address input, qrcode functionality
This would make Send screen just a handler for each step